lib/puppet/settings.rb

```diff
@@ -1156,11 +1156,22 @@ def add_environment_resources(catalog, sections)
     if configured_environment == "production" && envdir && Puppet::FileSystem.exist?(envdir)
       configured_environment_path = File.join(envdir, configured_environment)
       if !Puppet::FileSystem.symlink?(configured_environment_path)
-        catalog.add_resource(
-          Puppet::Resource.new(:file,
-                               configured_environment_path,
-                               :parameters => { :ensure => 'directory' })
-        )
+        if @service_user_available and @service_group_available
```

Comment:
You'll want set the owner/group to the string `"service"` like we do here: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/defaults.rb#L79-L81. It's a special value meaning set the owner/group to the service account if it exists, otherwise set it to `root`. The service account is the value of `Puppet[:user]` which is typically `puppet`. This way the caller doesn't have to duplicate that logic, so you can remove this line here and the else block below.